src/server/transaction.h

```diff
@@ -188,6 +188,9 @@ class Transaction {
     renabled_auto_journal_.store(true, std::memory_order_relaxed);
   }
 
+  // Conclude with empty hop
```

Comment:
I suggest maybe reduce the scope of this PR to only "Conclude" changes and refactor blocking stuff separately.